indico/modules/events/registration/fields/sessions.py

```diff
@@ -0,0 +1,102 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+import json
+
+from flask import session
+from marshmallow import ValidationError, fields, validate, validates_schema
+from sqlalchemy.orm import joinedload
+
+from indico.core.db.sqlalchemy import db
+from indico.core.marshmallow import mm
+from indico.modules.events.registration.fields.base import RegistrationFormFieldBase
+from indico.modules.events.registration.models.registrations import RegistrationData
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.modules.events.sessions.models.sessions import Session
+from indico.util.i18n import _
+
+
+class SessionsFieldDataSchema(mm.Schema):
+    minimum = fields.Integer(load_default=0, validate=validate.Range(0))
+    maximum = fields.Integer(load_default=0, validate=validate.Range(0))
+    collapse_days = fields.Bool(load_default=False)
+
+    @validates_schema(skip_on_field_errors=True)
+    def validate_min_max(self, data, **kwargs):
+        if data['minimum'] and data['maximum'] and data['minimum'] > data['maximum']:
+            raise ValidationError('Maximum value must be less than minimum value.', 'maximum')
+
+
+def _format_block(block, tzinfo):
+    return ' - '.join(
+        [
+            block.start_dt.astimezone(tzinfo).strftime('%H:%M'),
+            block.end_dt.astimezone(tzinfo).strftime('%H:%M'),
```

Comment:
Could be i18n'd:

```suggestion
            format_interval(block.start_dt.astimezone(tzinfo), block.end_dt.astimezone(tzinfo), "Hm"),
```

https://babel.pocoo.org/en/latest/api/dates.html#babel.dates.format_interval